pints/tests/test_slice_stepout.py

```diff
@@ -0,0 +1,350 @@
+# -*- coding: utf-8 -*-
+#
+# Tests the basic methods of the Slice Sampling routine.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+
+import unittest
+import numpy as np
+
+import pints
+import pints.toy
+
+debug = False
+
+
+class TestSliceStepout(unittest.TestCase):
+    """
+    Tests the basic methods of the Slice Sampling with Stepout routine.
+
+    Please refer to the _slice_stepout.py script in ..\_mcmc\_slice_stepout.py
+    """
+    def test_initialisation(self):
+        """
+        Tests whether all instance attributes are initialised correctly.
+        """
+        # Create mcmc
+        x0 = np.array([2, 4])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        # Test attributes initialisation
+        self.assertFalse(mcmc._running)
+        self.assertFalse(mcmc._ready_for_tell)
+        self.assertFalse(mcmc._first_expansion)
+        self.assertFalse(mcmc._interval_found)
+        self.assertFalse(mcmc._set_l)
+        self.assertFalse(mcmc._set_r)
+        self.assertFalse(mcmc._init_left)
+        self.assertFalse(mcmc._init_right)
+
+        self.assertEqual(mcmc._current, None)
+        self.assertEqual(mcmc._current_log_pdf, None)
+        self.assertEqual(mcmc._current_log_y, None)
+        self.assertEqual(mcmc._proposed, None)
+        self.assertEqual(mcmc._l, None)
+        self.assertEqual(mcmc._r, None)
+        self.assertEqual(mcmc._temp_l, None)
+        self.assertEqual(mcmc._temp_r, None)
+        self.assertEqual(mcmc._k, None)
+        self.assertEqual(mcmc._j, None)
+        self.assertEqual(mcmc._fx_l, None)
+        self.assertEqual(mcmc._fx_r, None)
+
+        self.assertEqual(mcmc._m, 50)
+        self.assertEqual(mcmc._active_param_index, 0)
+
+    def test_first_run(self):
+        """
+        Tests the very first run of the sampler.
+        """
+        # Create log pdf
+        log_pdf = pints.toy.GaussianLogPDF([2, 4], [[1, 0], [0, 3]])
+
+        # Create mcmc
+        x0 = np.array([2., 4.])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        # Ask should fail if _ready_for_tell flag is True
```

Comment:
Don't change the internals in a test! When writing the test, you don't know about the internals, or this flag!
So do something like:
```
mcmc.ask()
mcmc.ask()
```
and if the docstrings etc. say the second call should fail then test for that.